src/crypto/hash.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2017 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { getTransactionBytes } from './../transactions/transactionBytes';
+
+/**
+ * @method getSha256Hash
+ * @param stringToSign
+ * @param format
+ *
+ * @return {string}
+ */
+export function getSha256Hash(stringToSign, format) {
+	if (typeof stringToSign === 'object') {
+		return naclInstance.crypto_hash_sha256(stringToSign);
+	}
+
+	if (typeof stringToSign === 'string' && format === 'utf8') {
+		const encoded = naclInstance.encode_utf8(stringToSign);
+		return naclInstance.crypto_hash_sha256(encoded);
+	}
+
+	if (typeof stringToSign === 'string' && format === 'hex') {
+		const encoded = naclInstance.from_hex(stringToSign);
+		return naclInstance.crypto_hash_sha256(encoded);
+	}
```

Comment:
Neater would be to nest the different string processors in one block:
```js
if (typeof stringToSign === 'string') {
    if (!['utf8', 'hex'].includes(format)) {
        throw new Error('Unsupported string format. Currently only `hex` and `utf8` are supported.');
    }
    const encoded = format === 'utf8'
        ? naclInstance.encode_utf8(stringToSign)
        : naclInstance.from_hex(stringToSign);
    return naclInstance.crypto_hash_sha256(encoded);
}
```
This also makes it easy to provide different errors for different problems (unsupported data type vs unsupported string format).